light/stateprovider.go

```diff
@@ -433,16 +428,15 @@ func (s *StateProviderP2P) consensusParams(ctx context.Context, height int64) (t
 						case <-ctx.Done():
 							return
 						case out <- params:
+							cancel()
 							return
 						}
 					}
 				}
 
 			}(peer)
 		}
-		sig := make(chan struct{})
-		go func() { wg.Wait(); close(sig) }()
-		return sig, nil
+		return nil
```

Comment:
@Kbhat1 what was the purpose of removing the signal channel from the retryAll function? I believe we need to have some sort of waitgroup with signaling that we use to manage the spawned goroutines, right?

Reply:
Check discord discussion, waitgroup is unncessary since we don't want to wait on ALL providers to return a result - just one.